### PR TITLE
[red-knot] a few metaclass cleanups

### PR DESCRIPTION
**Pull request by carljm**
Just cleaning up a few small things I noticed in post-land review.

### CODE REVIEW

**Review by MichaReiser — APPROVED** [review]

File: crates/red_knot_python_semantic/resources/mdtest/metaclass.md [hunk 1]
```diff
@@ -161,12 +162,16 @@ reveal_type(B.__class__)  # revealed: Literal[M]
...
+`type.__new__` arguments, we should return its return type.
 
 ```py
-class A(metaclass=1): ...
+def f(*args, **kwargs) -> int: ...
+
+class A(metaclass=f): ...
 
-reveal_type(A.__class__)  # revealed: Literal[1]
+# TODO should be `int`
```
Comment (AlexWaygood) [inline crates/red_knot_python_semantic/resources/mdtest/metaclass.md:173]: If the metaclass is a callable that returns the type `int`, that means the variable `A` created by the `class` statement will be an instance of the class `int`, which means `A.__class__` will be the `int` class itself, which means this revealed type should in fact be the meta-type of `int`, i.e...

```suggestion
# TODO should be `type[int]`
```

File: crates/red_knot_python_semantic/src/types.rs [hunk 2]
```diff
@@ -2100,8 +2100,10 @@ impl<'db> Class<'db> {
             };
 
             let Type::ClassLiteral(mut candidate) = metaclass else {
-                // If the metaclass is not a class, return it directly.
-                return Ok(metaclass);
+                // TODO: If the metaclass is not a class, we should verify that it's a callable
+                // which accepts the same arguments as `type.__new__` (otherwise error), and return
```
Comment (AlexWaygood) [inline crates/red_knot_python_semantic/src/types.rs:2104]: And also verify that it returns a type assignable to `Instance(type)`? I don't think we can realistically support `class` statements that don't create classes

**Review by charliermarsh — APPROVED** [review]
Thanks

File: crates/red_knot_python_semantic/resources/mdtest/metaclass.md (hunk 1 above)
Comment (carljm) [reply crates/red_knot_python_semantic/resources/mdtest/metaclass.md:173]: Ah yeah, good catch!

File: crates/red_knot_python_semantic/src/types.rs (hunk 2 above)
Comment (carljm) [reply crates/red_knot_python_semantic/src/types.rs:2104]: Honestly it probably wouldn't be that hard, but it's also likely not worth it.

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**